readme.md

```diff
@@ -14,109 +14,50 @@ $ npm install --save download
 
 ## Usage
 
-If you're fetching an archive you can set `extract: true` in options and
-it'll extract it for you.
-
 ```js
-var Download = require('download');
-
-new Download({mode: '755'})
-    .get('http://example.com/foo.zip')
-    .get('http://example.com/cat.jpg')
-    .dest('dest')
-    .run();
-```
-
-
-## API
-
-### new Download(options)
-
-Creates a new `Download` instance.
-
-#### options
+const fs = require('fs');
+const download = require('download');
 
-Type: `object`
+download('http://unicorn.com/foo.jpg', 'dist').then(() => {
+	console.log('done!');
+});
 
-Options for [`got`](https://github.com/sindresorhus/got) or the underlying [`http`](https://nodejs.org/api/http.html#http_http_request_options_callback)/[`https`](https://nodejs.org/api/https.html#https_https_request_options_callback) request can be specified,
-as well as options specific to the `download` module as described below.
+download('http://unicorn.com/foo.jpg').then(data => {
+	fs.writeFileSync('dist/foo.jpg', data);
+});
 
-##### options.extract
-
-Type: `boolean`  
-Default: `false`
-
-If set to `true`, try extracting the file using [decompress](https://github.com/kevva/decompress/).
-
-##### options.mode
-
-Type: `string`
-
-Set mode on the downloaded file, i.e `{mode: '755'}`.
+download('http://unicorn.com/foo.jpg').pipe(fs.createWriteStream('dist/foo.jpg'));
+```
 
-##### options.strip
 
-Type: `number`  
-Default: `0`
+## API
 
-Remove leading directory components from extracted files.
+### download(url, [destination], [options])
 
-### .get(url, [dest])
+Returns both a Promise for a buffer and a Duplex stream with [additional events](https://github.com/sindresorhus/got#streams).
```

Comment:
I would also linkify `Duplex stream` to the Node.js docs, as most people wouldn't know how it works.